lib/src/Uncertainty/Distribution/UniformOverMesh.cxx

```diff
@@ -0,0 +1,214 @@
+UniformOverMesh::UniformOverMesh()
+  : ContinuousDistribution()
+{
+  setName( "UniformOverMesh" );
+  setDimension( 1 );
+}
+
+/* Parameters constructor */
+UniformOverMesh::UniformOverMesh(const Mesh & mesh)
+  : ContinuousDistribution()
+{
+  setName( "UniformOverMesh" );
+  setMesh(mesh);
+}
+
+/* Comparison operator */
+Bool UniformOverMesh::operator ==(const UniformOverMesh & other) const
+{
+  if (this == &other) return true;
+  return (mesh_ == other.mesh_);
+}
+
+Bool UniformOverMesh::equals(const DistributionImplementation & other) const
+{
+  const UniformOverMesh* p_other = dynamic_cast<const UniformOverMesh*>(&other);
+  return p_other && (*this == *p_other);
+}
+
+/* String converter */
+String UniformOverMesh::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << UniformOverMesh::GetClassName()
+      << " name=" << getName()
+      << " dimension=" << getDimension()
+      << " mesh=" << mesh_
+      << " meshDomain" << meshDomain_
+      << " simplicesVolumes" << simplicesVolumes_
+      << " meshVolume" << meshVolume_
+      << " simplexSelection" << simplexSelection_
+      << " uniformSimplex" << uniformSimplex_
+      << " integrationAlgorithm" << integrationAlgorithm_;
+  return oss;
+}
+
+String UniformOverMesh::__str__(const String & ) const
+{
+  OSS oss(false);
+  oss << getClassName() << "(mesh = " << mesh_ << ")";
+  return oss;
+}
+
+/* Virtual constructor */
+UniformOverMesh * UniformOverMesh::clone() const
+{
+  return new UniformOverMesh(*this);
+}
+
+/* Compute the numerical range of the distribution given the parameters values */
+void UniformOverMesh::computeRange()
+{
+  setRange(Interval(mesh_.getLowerBound(), mesh_.getUpperBound()));
+}
+
+
+/* Get one realization of the distribution */
+Point UniformOverMesh::getRealization() const
+{
+  const UnsignedInteger index = simplexSelection_.getRealization()[0];
+  const Indices simplex(mesh_.getSimplex(index));
+  const Sample vertices(mesh_.getVertices().select(simplex));
+  return DistFunc::rUniformSimplex(vertices);
+}
+
+
+/* Get the PDF of the distribution, i.e. P(point < X < point+dx) = PDF(point)dx + o(dx) */
+Scalar UniformOverMesh::computePDF(const Point & point) const
+{
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+  Scalar pdf = 0.0;
+  if (meshDomain_.contains(point))
+    pdf = 1.0 / meshVolume_;
+  return pdf;
+}
+
+
+/* Get the CDF of the distribution, i.e. P(X <= point) = CDF(point) */
+Scalar UniformOverMesh::computeCDF(const Point & point) const
+{
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+  Scalar cdf = 0.0;
+  // Waiting for a better implementation
+  const Interval i(Point(getDimension(), -SpecFunc::MaxScalar), point);
```

Comment:
Are you certain that `i` is a correct naming here?

Reply:
i like interval ;-) Ok I will change it